internal/testing/setup/setup.go

```diff
@@ -117,13 +122,13 @@ func NewRecordReplayClient(ctx context.Context, t *testing.T, rf func(r *httprep
 		if err != nil {
 			t.Fatal(err)
 		}
-		cleanup = func() {
+		cleanup := func() {
 			if err := rec.Close(); err != nil {
 				t.Fatal(err)
 			}
 		}
 
-		return client, cleanup
+		return client, cleanup, state.Unix()
```

Comment:
On second thought, UnixNano is probably a better choice. Just in case someone tries to use the state twice in the same second.

Reply:
Done.